wolfSSL/src/chapter02.md

```diff
@@ -680,15 +1190,321 @@ Enable ECC SECP384R1 Single-Precision (SP) support. Only applies to `WOLFSSL_SP_
+#### WOLFSSL_MIN_RSA_BITS
+
+By default, wolfSSL restricts RSA key sizes to 1024-bits minimum. To allow the decoding of smaller, less secure RSA keys like 512-bit keys, you will need to add the compiler flag `-DWOLFSSL_MIN_RSA_BITS=512` to CFLAGS or CPPFLAGS or define it in your user-settings header.
+
+#### WOLFSSL_MODE_AUTO_RETRY_ATTEMPTS
+
+Used to limit the possibility of an infinite retry loop.
+
+#### WOLFSSL_MULTICAST
+
+DTLS multicast feature.
+
+#### WOLFSSL_MULTICAST_PEERS
+
+Multicast feature defined as max allowed 100 peers.
+
+#### WOLFSSL_NAMES_STATIC
+
+Uses static ECC structs for Position Independent Code (PIC).
+
+#### WOLFSSL_SEND_HRR_COOKIE
+
+TLS extension used by DTLS 1.3.
+
+#### WOLFSSL_SEP
+
+Feature certificate policy set extension.
+
+#### WOLFSSL_SESSION_ID_CTX
+
+Used to copy over application session context ID.
+
+#### WOLFSSL_SESSION_TIMEOUT
+
+Default session resumption cache timeout in seconds is used to define timeout manually.
+
+#### KEEP_OUR_CERT
+
+Used to ensure the ability to return SSL certificate.
+
+#### KEEP_PEER_CERT
+
+Retains peer certs. Parts of the OpenSSL compatibility layer require peer certs.
+
+#### WOLFSSL_SIGNER_DER_CERT
+
+Used to no CERT DER buffer OR NULL cm. Used to allocate DER and cert.
+
+#### CA_TABLE_SIZE
+
+wolfSSL certificate manager. CA signer table. Used in certificate manager and be defined size.
+
+#### ECDHE_SIZE
+
+Define to allow this to be overridden at compile-time. ECDHE server size defaults to 256 bits, which can set a predetermined ECDHE curve size. The default is 32 bytes.
+
+#### CIPHER_NONCE
+
+It is used as a cryptographic number, which is implemented in authentication. It is a pseudo-random number, which is an integrity-only cipher suite.
+
+#### WOLFSSL_USE_POPEN_HOST
+
+Uses popen for creating socket with host and post for wolfio.c socket open code used with CRL and OCSP.
+
+#### CloseSocket
+
+A way to override the function used for closing a socket. Used with CRL, OCSP and BIO.
+
+#### CONFIG_POSIX_API
+
+Enables POSIX names for networking systems calls.
+
+#### WOLFSSL_USER_CURRTIME
+
+Add in the option to use in test.h without the gettimeofday function using the macro `WOLFSSL_USER_CURRTIME`.
+
+#### WOLFSSL_USER_MUTEX
+
+Option for user-defined mutexes with `WOLFSSL_USER_MUTEX`.
+
+#### DEFAULT_MIN_ECCKEY_BITS
+
+Identifies the minimum number of bits in ECCkey.
+
+#### DEFAULT_MIN_RSAKEY_BITS
+
+Identifies the minimum number of bits in RSA key.
+
+#### EXTERNAL_SERIAL_SIZE
+
+A raw serial number byte that writes X509 serial numbers in unsigned binary to a buffer. For all cases, the buffer needs to be at least `EXTERNAL_SERIAL_SIZE` (32). On success, it returns `WOLFSSL_SUCCESS`. Note: this is a internal macro that cannot be user defined.
+
+#### LARGE_STATIC_BUFFERS
+
+Embedded callbacks require large static buffers; make sure it gives the option to enable larger buffers to 16K.
+
+#### LIBWOLFSSL_VERSION_STRING
+
+A const char pointer defining the version.
```

Comment:
This is the wolfSSL version string populated for release bundles or when ./configure is run. There is also a 32-bit HEX version of this in `LIBWOLFSSL_VERSION_HEX`. These come from `wolfssl/version.h`.